src/net/sourceforge/kolmafia/utilities/FileUtilities.java

```diff
@@ -565,4 +567,14 @@ public static String getRelativePath(File home, File f) {
 
     return s;
   }
+
+  public static boolean isEmptyDirectory(Path path) throws IOException {
+    if (!Files.isDirectory(path)) return false;
+
+    var listFiles = Files.list(path);
```

Comment:
Not certain of the syntax but I believe you.  Any particular benefit to this instead of checking the length of Files.list(path)?  I do note that Files.list() won't return the dot or dot dot references which is the only weird case I can think of.  Obviously not blocking.

Reply:
`Files.list` returns a stream. Checking whether the stream has any values is faster than checking whether the length of the list is 0, because checking the length will enumerate every item in the list, which /could/ be slow (e.g. on HDD, or if a script somehow has hundreds or thousands of files).